introduce no-JIT and mpi-np != 2 runs on CI

slayoo opened the pull request.

david-zwicker commented:
The number of Checks got a little bit out of hands :) I think we need to somehow reduce this to also keep the used resources in check. I propose to test not all combinations of settings.

slayoo commented:
In #137, we have introduced an `mpi-np` matrix from this PR (with settings for `mpiexec -np N`). Previously, everything was tested on 2 processes, now we do both 2 and 3. This helps to detect issues such as #121 (now fixed). Let's stick to this for now, until we find a good case for introducing other combinations - agree with @david-zwicker that we should try to keep the resouce usage and time-to-complete of the CI runs within reasonable limits.
